### PR TITLE
Created User Model of type PFUser, added tap gesture to remove keyboa…

### PR DESCRIPTION
**Pull request by ristau**
…rd on login, implemented AppDelegate Notifications for managing segues based on login status, created outlets on profile view controller and linked to the user from Parse

### CODE REVIEW

**Review by mileung — APPROVED** [review]

File: LinkBase/LinkBase/CreateAccountViewController.swift [hunk 1]
```diff
@@ -28,39 +30,57 @@ class CreateAccountViewController: UIViewController {
...
+        
+
+        createUser{ (success) -> Void in
+            
+            if success {
+                
+                currentUser!.signUpInBackground { (succeeded: Bool, error: Error?) in
+                 if let error = error {
+                     print("error: \(error)")
+                    //				self.alert(message: "Username already taken")
```
Comment (luckyx1) [inline LinkBase/LinkBase/CreateAccountViewController.swift:63]: we should use the alert from the lab we did

File: LinkBase/LinkBase/User.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    static let userDidLogIn = "UserDidLogIn"
+    
+    convenience init(_email: String, _password: String, firstname: String, lastname: String){
+        
+        self.init()
+        
+        username = _email
+        email = _email
+        password = _password
+        add(firstname as Any, forKey: "firstname")
```
Comment (luckyx1) [inline LinkBase/LinkBase/User.swift:28]: I don't like the AnyKeyword, we should truncate to String for this

File: LinkBase/LinkBase/User.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        add(lastname as Any, forKey: "lastname")
```
Comment (luckyx1) [inline LinkBase/LinkBase/User.swift:29]: same here

File: LinkBase/LinkBase/UtilityFunctions.swift [hunk 4]
```diff
@@ -46,3 +46,27 @@ func getDateFromPriorWeek(currentDate: Date, numWeeks: Int) -> Date {
  
     return newDate
 }
+
+
+// this method removes the strings from the first and last name properties in Parse
+func getUserFirstName(user: User) -> String {
```
Comment (luckyx1) [inline LinkBase/LinkBase/UtilityFunctions.swift:52]: then we don't need these right?

**Review by luckyx1 — COMMENTED** [review]
A few changes requested, but if you comment why such is such, we can merge

**Comment by ristau** [comment]
1.  The first name/last name stuff is now cleaned up.  Basically, when you add an object to Parse using the "add" method, it creates an array, even if the input is a String (I had actually changed from Any to String).  But it still kept requiring an array.  I had to delete all user records, as well as update the keys in the Schema.  This now works properly, and was updated in the PR #23.

2.  Login: Yes, we should add alerts.  I suppressed them in the alert box but kept them in the console, because the alert was not reflecting the actual error.  I would like to focus first on getting the data models in this section implemented, so would recommend these types of fixes be prioritized after the data models are up and running.

**Pull request merged**